### DISCUSSION TITLE
Adding Sensors to tasmota32-webcam

### DISCUSSION THREAD

**Discussion by czarnikjak** [arendst/Tasmota#13652] [Q&A] [answered]
Hi,

I am trying to add RC_SWITCH to the Webcam Firmware.

I managed to compile the binary based on 10.0.2 Webcam firmware by adding #define USE_RC_SWITCH  to tasmota_configurations_ESP32.h ( #ifdef FIRMWARE_WEBCAM section).

I also had to add lib/lib_rf to lib_extra_dirs in platformio_tasmota_env32.ini, 
but the build compiles and flashes onto my ESP32-CAM just fine.

Compiler shows this line during the compilations, hence i think that RC_Switch is being added:
Retrieved `.pio/build/tasmota32-webcam/libe06/rc-switch/RCSwitch.cpp.o' from cache

The problem is, when i go to configuration and try to assign RFSend or RFRecv to one of the GPIOs the options are not there.
Am i missing something simple here? (RFProtocol command is also returning "Command":"Unknown")

Apologies if this is an obvious question, its first time i am building my own Tasmota build.

Thanks

**Comment by barbudor** [comment, marked as answer]
You are doing many things wrong
Please again https://tasmota.github.io/docs/Compile-your-build/

As a general rule you can't add anything to a predefined variant and changing tasmota files like you did is wrong. Only overrides files should be changed to customize 'tasmota/tasmota32' 

Specifically Webcam firmware is very specific and very tight in term of capacity. Although you may be able to successfully build webcam+rc-switch, we will not support you on this very specific case.

**Reply by czarnikjak** [reply]
Thank you for your quick response. I will start from scratch and only modify override files.

Interesting that you say the Webcam firmware is very tight in terms of Capacity. What do you mean by that exactly? Its size in Flash is only about 1MB, smaller than standard tasmota32. RAM also doesn't appear to be a problem on mine with 93KB free(ES32-Cam module also adds 4MB of PSRAM). What is the constraint?

**Reply by barbudor** [reply]
Essentially CPU

**Reply by czarnikjak** [reply]
Got ya.
I assume RFrecv is also CPU intensive?
Ive managed to successfully build it now and RFrecv is showing under Configuration and I am able to assign it to a GPIO.

However, with RFrecv enabled, wcinit results in reboot of the device with the following error message:
_{"Info3":{"RestartReason":{"Exception":28,"Reason":"LoadProhibited","EPC":"401aaf0b"......._

Is that due to CPU resource exhaustion?

**Reply by barbudor** [reply]
Not necesseraly
"Load Prohibited" means that the CPU tried to read (load) from a unauthorized memory address.
